### PR TITLE
client-go: move out the judgment logic from loop for reduce the count…

### PR DESCRIPTION
**Pull request by weilaaa**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

- staging/src/k8s.io/client-go: move out the judgment logic from loop for reduce the count of judgment

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
```
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```

```
/assign @smarterclayton @caesarxuchao

### CODE REVIEW

File: staging/src/k8s.io/client-go/rest/request.go [hunk 1]
```diff
@@ -365,10 +365,10 @@ func (r *Request) SpecificallyVersionedParams(obj runtime.Object, codec runtime.
 		r.err = err
 		return r
 	}
+	if r.params == nil {
+		r.params = make(url.Values)
+	}
 	for k, v := range params {
-		if r.params == nil {
```
Comment (yliaog) [inline staging/src/k8s.io/client-go/rest/request.go:369]: there is a change in behavior. previously, if params is empty, r.params will be left untouched, i.e. nil.

**Comment by weilaaa** [comment]
thanks for point it out, I would close this PR

**Pull request closed without merging**